### PR TITLE
Tehseen/updatedlengthsagain

### LINKED ISSUE #28: Package lengths are wrong
src/BusinessLayer/SerialReporting.cpp
Lines 25, 26 and 28, according to the communications protocol, they should be 7, 51, and 3 respectively

### PR DESCRIPTION
**Pull request by bill-luu**
Fixes #28

### CODE REVIEW

**Comment by enochtsang** [comment]
Haha the commits and the pull requester don't match up.

**Comment by bill-luu** [comment]
Oh shoot can you also fix up the expected values in Epsilon-Telemetry-Test-Tool/src/Tests/BusinessLayer/SerialReportingTest.cpp

**Comment by bill-luu** [comment]
@enochtsang can you explain what this Travis Error is? I'm still not too familiar with google testing stuff.

**Comment by bill-luu** [comment]
@tehseeni please fix the merge conflicts

**Comment by enochtsang** [comment]
Please fix the build.

**Comment by enochtsang** [comment]
This is actually a very useful failure, not a BS update error or anything. 

The new changes that Tehseen made are untested, we know this because the tests have not been updated. @tehseeni I'll challenge you to update the test as well, you can message me directly if you have any questions.

File: src/Tests/BusinessLayer/SerialReportingTest.cpp [hunk 1]
```diff
@@ -631,6 +631,7 @@ TEST_F(SerialReportingTest, sendLightsTest) // TODO create function which build
     const unsigned int payloadLength = expectedPackageLength - COBS_ADDITIONAL_FRAME_DATA_SIZE;
     unsigned char data[payloadLength];
     data[0] = CcsDefines::LIGHTS_PKG_ID;
+    data[1] = CcsDefines::LIGHTS_ALIVE;
```
Comment (bill-luu) [inline src/Tests/BusinessLayer/SerialReportingTest.cpp]: There's no `LIGHTS_ALIVE` in `CcsDefines.h`. Lights Alive would be a boolean value. Check out how the battery_alive is represented in `BatteryData.h` and `SendBattery()`.

File: src/BusinessLayer/SerialReporting.cpp [hunk 2]
```diff
@@ -307,11 +307,11 @@ void SerialReporting::sendBattery()
...
-    packetPayload[46] = batteryData_.highCellVoltageId;
-    writeUShortIntoArray(packetPayload, 47, batteryData_.averageCellVoltage);
-    packetPayload[49] = (unsigned char)batteryData_.prechargeState;
-    packetPayload[50] = batteryData_.auxVoltage;
-    bool auxBmsaliveArray[] = {batteryData_.auxBmsAlive};
+    packetPayload[45] = batteryData_.highCellVoltageId;
+    writeUShortIntoArray(packetPayload, 46, batteryData_.averageCellVoltage);
+    packetPayload[48] = (unsigned char)batteryData_.prechargeState;
+    packetPayload[49] = batteryData_.auxVoltage;
+    bool auxBmsaliveArray[] = {packetPayLoad, 50,batteryData_.auxBmsAlive};
```
Comment (bill-luu) [inline src/BusinessLayer/SerialReporting.cpp]: `packetPayload`, not `packetPayLoad`.

File: src/BusinessLayer/SerialReporting.cpp [hunk 3]
```diff
@@ -353,7 +353,8 @@ void SerialReporting::sendLights()
     const unsigned int unframedPacketLength = LIGHTS_LENGTH + CHECKSUM_LENGTH;
     unsigned char packetPayload[unframedPacketLength];
     packetPayload[0] = CcsDefines::LIGHTS_PKG_ID;
-    packetPayLoad[1] = CcsDefines::LIGHTS_ALIVE;
+    bool lightsAliveArray[] = {lightsData_.alive};
```
Comment (bill-luu) [inline src/BusinessLayer/SerialReporting.cpp]: You'll have to edit the actual `lightsData.h` file to get it to work. `alive` hasn't been defined yet.

**Comment by bill-luu** [comment]
Reassigning, as Tehseen has her hands full with Tab & Battery pages in the debug display.

**Comment by bill-luu** [comment]
Check the build, the SerialReportingTest function is also not working. The fix that you implemented in SerialReporting needs to be implemented there too.

**Comment by Reddragoneer1** [comment]
👍

**Pull request merged**